src/status_im2/contexts/add_new_contact/events.cljs

```diff
@@ -23,50 +18,73 @@
+                          (if-some [[_ match] (re-matches regex $)]
+                            match
+                            $)
+                          (if (empty? $) nil $))
+        public-key?     (validators/valid-public-key? id)
+        compressed-key? (validators/valid-compressed-key? id)
+        type            (cond (empty? id)     :empty
+                              public-key?     :public-key
+                              compressed-key? :compressed-key
+                              :else           :ens-name)
+        ens-name        (when (= type :ens-name)
+                          (stateofus/ens-name-parse id))]
+    {:input    input
+     :id       id
+     :type     type
+     :ens-name ens-name}))
 
 (rf/defn set-new-identity
   {:events [:contacts/set-new-identity]}
   [{:keys [db]} input]
-  (let [input           (utils/safe-trim input)
-        public-key?     (validators/valid-public-key? input)
-        compressed-key? (validators/valid-compressed-key? input)
-        ens-name        (if compressed-key? nil (stateofus/ens-name-parse input))
-        on-success      (fn [pubkey]
-                          (rf/dispatch
-                           [:contacts/set-new-identity-success
-                            input ens-name pubkey]))
-        on-error        (fn [err]
-                          (rf/dispatch
-                           [:contacts/set-new-identity-error err input]))]
-    (cond
-      (empty? input) {:db (dissoc db :contacts/new-identity)}
-      public-key?    {:db (assoc db
-                                 :contacts/new-identity
-                                 {:input      input
-                                  :public-key input
-                                  :ens-name   nil
-                                  :state      :error
-                                  :error      :uncompressed-key})}
-      :else
-      (cond->
-        {:db (assoc db
-                    :contacts/new-identity
-                    {:input      input
-                     :public-key nil
-                     :ens-name   nil
-                     :state      :searching
-                     :error      nil})}
-        compressed-key?       (assoc :contacts/decompress-public-key
-                                     {:public-key input
-                                      :on-success on-success
-                                      :on-error   on-error})
-        (not compressed-key?) (assoc :contacts/resolve-public-key-from-ens-name
-                                     {:chain-id   (ethereum/chain-id db)
-                                      :ens-name   ens-name
-                                      :on-success on-success
-                                      :on-error   on-error})))))
+  (let [{:keys [input id type ens-name]} (identify-type input)]
+    (case type
+      :empty          {:db (dissoc db :contacts/new-identity)}
+      :public-key     {:db (assoc db
```

Comment:
This would seem like a regression, but it is intended behavior. For example, the "This is not a chatkey" is in the figma, and @cammellos and I both worked specifically on removing support for uncompressed keys.

Reply:
I really remember someone telling me nodes will always support uncompressed and compressed keys, and so should the clients. I'm confused, when I look at the specs in https://specs.status.im/spec/2#public-key-serialization, there's nothing saying uncompressed keys should be avoided or not supported when implementing clients. I'm probably missing something.